tests/migrations/test_a4bf1f58ce69.py

```diff
@@ -0,0 +1,136 @@
+            journalist = self.session.query(User).filter_by(id=journalist_id).one()
+            source_id = random.randint(1, self.NUM_SOURCES)
+            add_reply(self.session, journalist.uuid, source_id)
+
+        self.session.commit()
+
+    def check_upgrade(self):
+        """
+        Make sure each reply in the replies table has the correct journalist_id stored for the
+        associated journalist by making sure a User account exists with that journalist id.
+        """
+        replies = self.session.query(Reply).all()
+        assert len(replies)
+
+        for reply in replies:
+            # Will fail if User does not exist
+            self.session.query(User).filter_by(id=reply.journalist_id).one()
+
+        self.session.close()
+
+
+class DowngradeTester:
+    """
+    Verify that downgrading from the target migration keeps in place the updates from the migration
+    since there is no need to add bad data back into the db (the migration is backwards compatible).
+    """
+
+    NUM_USERS = 20
+    NUM_SOURCES = 20
+    NUM_REPLIES = 40
+
+    def __init__(self, homedir):
+        subprocess.check_call(["sqlite3", os.path.join(homedir, "svs.sqlite"), ".databases"])
+        self.session = db.make_session_maker(homedir)()
+
+    def load_data(self):
+        """
+        Load data that has the bug where user.uuid is stored in replies.journalist_id.
+        """
+        for _ in range(self.NUM_SOURCES):
+            add_source(self.session)
+
+        self.session.commit()
+
+        for i in range(self.NUM_USERS):
+            if i == 0:
+                # As of this migration, the server tells the client that the associated journalist
+                # of a reply has been deleted by returning "deleted" as the uuid of the associated
+                # journalist. This gets stored as the jouranlist_id in the replies table.
+                #
+                # Make sure to test this case as well.
+                add_user(self.session, "deleted")
+                journalist = self.session.query(User).filter_by(uuid="deleted").one()
+                source_id = random.randint(1, self.NUM_SOURCES)
+                add_reply(self.session, journalist.id, source_id)
+            else:
+                add_user(self.session)
+
+        self.session.commit()
+
+        # Add replies from randomly-selected journalists to a randomly-selected sources
+        for _ in range(1, self.NUM_REPLIES):
+            journalist_id = random.randint(1, self.NUM_USERS)
+            source_id = random.randint(1, self.NUM_SOURCES)
+            add_reply(self.session, journalist_id, source_id)
+
+        self.session.commit()
+
+    def check_downgrade(self):
+        """
+        Make sure each reply in the replies table has the correct journalist_id stored for the
+        associated journalist by making sure a User account exists with that journalist id.
+        """
+        replies = self.session.query(Reply).all()
```

Comment:
This does verify that all replies have valid journalist IDs, but not that replies weren't changed by the downgrade. I feel a little silly suggesting a more rigorous check for a `pass` downgrade, but if we are going to test it, perhaps you could stash the `id` and `journalist_id` for all replies at the end of `load_data`, then iterate `self.replies` here, so you're confirming that after the downgrade you have the same set of replies, associated with the same journalists.

Reply:
I think it does test that the replies did not change, because if the downgrade updated the ids back to uuids then the check: `self.session.query(User).filter_by(id=reply.journalist_id).one()` would fail.

Reply:
I don't want to belabor this, because again the downgrade doesn't do anything, but if a reply's journalist ID were changed by the downgrade but was still a valid ID, this check wouldn't catch it. Never mind; are we going to check all the other data too to make sure the no-op downgrade didn't touch it? :smile: 

We probably ought to talk about whether we want to have any tests at all for migrations that don't do anything. They make the tests take longer without adding any surety about the changes.

Reply:
Update: After some back and forth with @rmol on this, I went ahead and addressed this by passing on the downgrade test scenario. @rmol helped me realize that the test would pass even without the bug fix in place -- it's not the regression test that I thought it was. Since we can see in the code that we pass on downgrade, we know there's nothing to test, so I agree that it's perfectly fine to skip this.